scripts/floods/earthengine_image_test.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for earthengine_image.py"""
+
+import os
+import sys
+import unittest
+import ee
+
+# Allows the following module imports to work when running as a script
+sys.path.append(
+    os.path.dirname(os.path.dirname(
+            os.path.abspath(__file__))))
+
+import earthengine_image as eei
+
+_TESTDIR = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                                                'test_data')
+
+class EarthengineImageTest(unittest.TestCase):
+  def setUp(self):
+    ee.Initialize()
```

Comment:
interesting.  does the test not need auth?

Reply:
No, the test only generates an image config that includes all parameters to generate an image and compares that with the expected image parameters.

To actually generate an image, it must be submitted for an export which requires ee.Authenticate() that would prompt the user. The tests don't call export.